modules/apps/29-fee/ibc_module_test.go

```diff
@@ -468,3 +470,301 @@ func (suite *FeeTestSuite) TestOnChanCloseConfirm() {
+					TimeoutFee: validCoins3,
+				},
+				refundAcc.String(),
+				[]string{},
+			)
+			err := suite.chainA.GetSimApp().IBCFeeKeeper.EscrowPacketFee(suite.chainA.GetContext(), identifiedFee)
+			suite.Require().NoError(err)
+
+			// set up module and callbacks
+			module, _, err := suite.chainA.App.GetIBCKeeper().PortKeeper.LookupModuleByPort(suite.chainA.GetContext(), ibctesting.TransferPort)
+			suite.Require().NoError(err)
+
+			cbs, ok := suite.chainA.App.GetIBCKeeper().Router.GetRoute(module)
+			suite.Require().True(ok)
+
+			// malleate test case
+			tc.malleate(suite)
+
+			result := cbs.OnRecvPacket(suite.chainA.GetContext(), packet, refundAcc)
+
+			if tc.sourceRelayer {
+				suite.Require().NoError(err, "unexpected error for case: %s", tc.name)
+				ack := types.IncentivizedAcknowledgement{
+					Result:                channeltypes.NewResultAcknowledgement([]byte{1}).Acknowledgement(),
+					ForwardRelayerAddress: suite.chainB.SenderAccount.GetAddress().String(),
+				}
+				suite.Require().Equal(result, ack)
+			} else {
+				suite.Require().NoError(err, "unexpected error for case: %s", tc.name)
+				ack := types.IncentivizedAcknowledgement{
+					Result:                channeltypes.NewResultAcknowledgement([]byte{1}).Acknowledgement(),
+					ForwardRelayerAddress: "",
+				}
+				suite.Require().Equal(result, ack)
+			}
+		})
+	}
+}
+
+func (suite *FeeTestSuite) TestOnAcknowledgementPacket() {
+	testCases := []struct {
+		name     string
+		malleate func(suite *FeeTestSuite)
+		expPass  bool
+	}{
+		{
+			"success",
+			func(suite *FeeTestSuite) {
+				packetId := channeltypes.NewPacketId(suite.path.EndpointA.ChannelID, suite.path.EndpointA.ChannelConfig.PortID, suite.chainA.SenderAccount.GetSequence())
+				identifiedFee := types.NewIdentifiedPacketFee(
+					packetId,
+					types.Fee{
+						ReceiveFee: validCoins,
+						AckFee:     validCoins2,
+						TimeoutFee: validCoins3,
+					},
+					suite.chainA.SenderAccount.GetAddress().String(),
+					[]string{},
+				)
+				err := suite.chainA.GetSimApp().IBCFeeKeeper.EscrowPacketFee(suite.chainA.GetContext(), identifiedFee)
+				suite.Require().NoError(err)
+			},
+			true,
+		},
+		{
+			"identified packet fee doesn't exist",
+			func(suite *FeeTestSuite) {
+				suite.Require().Error(types.ErrFeeNotFound)
+			},
+			false,
+		},
+		{
+			"refundAccount doesn't exist",
+			func(suite *FeeTestSuite) {
+				packetId := channeltypes.NewPacketId(suite.path.EndpointA.ChannelID, suite.path.EndpointA.ChannelConfig.PortID, suite.chainA.SenderAccount.GetSequence())
+				identifiedFee := types.NewIdentifiedPacketFee(
+					packetId,
+					types.Fee{
+						ReceiveFee: validCoins,
+						AckFee:     validCoins2,
+						TimeoutFee: validCoins3,
+					},
+					suite.chainB.SenderAccount.GetAddress().String(),
+					[]string{},
+				)
+				suite.chainA.GetSimApp().IBCFeeKeeper.EscrowPacketFee(suite.chainA.GetContext(), identifiedFee)
+				suite.Require().Error(types.ErrRefundAccNotFound)
```

Comment:
I think what you have here is the equivalent to not calling `EscrowPacketFee` at all. By passing in a refund account that doesn't exist `EscrowPacketFee` will error and do nothing. Maybe I'm missing something? :)

Reply:
true, i'll update the test to include a wrong ack format